pennylane/ops/functions/evolve.py

```diff
@@ -58,7 +65,28 @@ def evolve(op: Union[Operator, ParametrizedHamiltonian]):
 
     Please check the :class:`.ParametrizedEvolution` class for more information.
     """
-    if isinstance(op, ParametrizedHamiltonian):
-        return ParametrizedEvolution(H=op)
+    return ParametrizedEvolution(H=op)
+
 
-    return Evolution(generator=op, param=1.0)
+@evolve.register
+def evolution(op: Operator, coeff: float = 1):
```

Comment:
Should we add `do_queue` and `id` here? It seems we have them on some operator constructors (like `sum`, `sprod`, `prod`, `pow`) and not others (for example `ctrl` has neither, `exp` has only `id`). Do you know why?

Reply:
I've never seen anyone using these kwargs. Moreover, I'd avoid using `do_queue` because we want to hide the queueing behaviour from the user. Regarding the `id` kwarg, I have no idea what's the use case for it. Maybe @josh146 or @albi3ro know.

Reply:
- `do_queue` is there historically, as we occasionally needed ways of instantiating operators _without_ queueing them. I believe `stop_recording()` is the official replacement for `do_queue`.

- `id` is there to allow users to 'name' a specific operation, in order to track it during transformations etc. It is important for characterisation, for example in the `qml.fourier` module. @mariaschuld could potentially add more context here.

It might make sense to formalize the reason behind `id` existing 🤔

Reply:
@dwierichs I believe you may have also in the past come across use-cases for `id`?

Reply:
Yes indeed. I _think_ in all cases so far I ended up not using it, but I'm not entirely sure.
I could imagine that similar functionality could be achieved with the `inspect` module and argument tracing, but this @mariaschuld will know better :)